etc/generate-docs.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# usage: ./etc/generate-docs.sh [new version string]
+
+# exit if any command fails
+set -e
+
+if ! command -v jazzy > /dev/null; then
+  gem install jazzy || { echo "ERROR: Failed to locate or install jazzy; please install yourself with 'gem install jazzy' (you may need to use sudo)"; exit 1; }
+fi
+
+version=${1}
```

Comment:
Actually, can we ensure this version is non-empty? I'm worried the commit might happen without a version string below.

Reply:
good point, I added a check for this.